internal/gcsx/client_readers/gcs_reader.go

```diff
@@ -127,11 +137,13 @@ func (gr *GCSReader) ReadAt(ctx context.Context, p []byte, offset int64) (gcsx.R
 	if readerType == RangeReaderType {
 		readerResponse, err = gr.rangeReader.ReadAt(ctx, readReq)
 		gr.totalReadBytes += uint64(readerResponse.Size)
+		gr.updateExpectedOffset(offset + int64(readerResponse.Size))
```

Comment:
Should we do this as part of defer statement? - As we always want to update the expected offset.